Library/Homebrew/unpack_strategy/dmg.rb

```diff
@@ -41,7 +41,10 @@ def initialize(path)
       # Check if path is considered disk image metadata.
       sig { params(pathname: Pathname).returns(T::Boolean) }
       def self.dmg_metadata?(pathname)
-        DMG_METADATA.include?(pathname.cleanpath.ascend.to_a.last.to_s)
+        pathname_string = pathname.cleanpath.ascend.to_a.last.to_s
+        return true if /^.HFS\+ Private+/.match?(pathname_string)
```

Comment:
I think it'd be nicer to add this to `DMG_METADATA` and refactor it to not use `%w`